get_github_coderabbit_review/coderabbit_fetcher/processors/thread_processor.py

```
@@ -0,0 +1,511 @@
+        Args:
+            thread_comments: List of comments in chronological order
+            
+        Returns:
+            ThreadContext object with analyzed thread information
+            
+        Raises:
+            CommentParsingError: If thread cannot be processed
+        """
+        try:
+            if not thread_comments:
+                raise CommentParsingError("Empty thread provided")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)